uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/pom/PomEditorDefault.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.guvnor.structure.backend.pom;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.guvnor.structure.pom.DynamicPomDependency;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.java.nio.file.Files;
+import org.uberfire.java.nio.file.Paths;
+import org.uberfire.java.nio.file.StandardOpenOption;
+
+public class PomEditorDefault implements PomEditor {
+
+    private static final String DELIMITER = ":";
+    private final Logger logger = LoggerFactory.getLogger(PomEditorDefault.class);
+
+    private MavenXpp3Reader reader;
+    private MavenXpp3Writer writer;
+
+    public PomEditorDefault() {
+        reader = new MavenXpp3Reader();
+        writer = new MavenXpp3Writer();
+    }
+
+    public boolean addDependency(DynamicPomDependency dep,
+                                 Path pomPath) {
+        if (dep.getGroupID().isEmpty() || dep.getArtifactID().isEmpty()) {
+            return false;
```

Comment:
this line is not covered by any tests, consider adding a test for this case

Reply:
Added more test and added additional checks about null and empty params